department-of-veterans-affairs/caseflow
# TYLERB/APPEALS-10820: HLR SC Permissions Change Banner Creation

Resolves [APPEALS-10820](https://vajira.max.gov/browse/APPEALS-10820)

### Description
When completing Step 3 (Review screen) for Higher-Level Review and Supplemental Claim intake, non-VHA business line users will now be shown an informational banner informing them that only users that are VHA members can submit a HLR/SC intake using the VHA benefit type.

### Acceptance Criteria
- [x] Banner informational text
  - [x] Banner header: HLR and SC Permissions Update
  - [x] Banner text: Only VHA team members can establish Higher Level Reviews and Supplemental Claims with VHA issues. If you have a VHA claim, please return the packet to the main "VHA" queue in the Centralized Mail Portal or send downloaded documents to [VHABENEFITAPPEALS@va.gov](mailto:VHABENEFITAPPEALS@va.gov).
- [x] Proper display of banner on SC and HLR with the correct user
  - [x] Supplemental Claims
    - [x] The banner displays for VHA business line members
    - [x] The banner does not display for users that are not members of the VHA business line organization
  - [x] Higher Level Reviews
    - [x] The banner displays for VHA business line members
    - [x] The banner does not display for users that are not members of the VHA business line organization
  - [x] The banner does not appear on Appeal intakes
- [x] Clicking on the banner Email link opens up the local email client with the following information:
  - [x] Auto-Populated Recipient: [VHABENEFITAPPEALS@va.gov](mailto:VHABENEFITAPPEALS@va.gov) 
  - [x] Auto-Populated Subject: Potential VHA Higher-Level Review or Supplemental Claim

### Testing Plan
1. Login with a regular intake user BVAISHAW will work
2. Navigate to the /intake page
3. Start a higher level review intake
4. Verify that the banner is present on the page and that it matches the designs https://bipuiux.invisionapp.com/console/share/QTCKRDXAHEN/953715694
![image](https://user-images.githubusercontent.com/109369527/202564454-274332d6-538d-48f4-82b8-9851895d3fb8.png)
5. Click the email link with the text VHABENEFITAPPEALS@va.gov in the banner
6. Verify that the email link opens up your native email application.
7. Verify that the email link auto populates the to field with VHABENEFITAPPEALS@va.gov
8. Verify that the email link auto populates the subject line with Potential VHA Higher-Level Review or Supplemental Claim
![image](https://user-images.githubusercontent.com/109369527/202565175-021af283-6411-43ed-8d7f-ac914e277df5.png)

9. Cancel the intake and start over with a supplemental claim intake
10. Verify that the banner is present on the page and the email functionality works.
11. Cancel the intake and start over with an appeal intake
12. Verify that the banner is not present on the page
13. Login with an intake user that is also a member of the VHA business line or create one with the rails console
```ruby
BusinessLine.find_by(url: "vha").add_user(FactoryBot.create(:user, roles:['Admin Intake'], css_id:'TESTVHAUSER'))
```
14. Start a higher level review intake with the new user or your chosen existing user
15. Verify that the banner is not present on the review page
![image](https://user-images.githubusercontent.com/109369527/202566127-a749ffa8-9bb3-4f50-856a-9299ef6d1a2c.png)
16. Cancel the intake and start a new supplemental claim intake
17. Verify that the banner is not present on the review page
18. Cancel the intake and start a new appeal intake
19. Verify that the banner is not present on the review page
